src/python/dxpy/nextflow/nextflow_templates.py

```diff
@@ -66,11 +66,13 @@ def get_nextflow_src(custom_inputs=None, profile=None, resources_dir=None):
     with open(os.path.join(str(get_template_dir()), get_source_file_name()), 'r') as f:
         src = f.read()
 
+    exclude_input_download=""
     applet_runtime_params = ""
     for i in custom_inputs:
         value = "${%s}" % (i['name'])
         if i.get("class") == "file":
             value = "dx://${DX_WORKSPACE_ID}:/$(echo ${%s} | jq .[$dnanexus_link] -r | xargs -I {} dx describe {} --json | jq -r .name)" % i['name']
+            exclude_input_download+="--except {} ".format(i['name'])
```

Comment:
dx-download-all-inputs should exclude those file type input from schema, since those are referred as dx url by nextflow-io later, while the soft configs and parameter file need to be localized when passing to `nextflow run`